core/src/test/java/org/infinispan/stream/BaseStreamTest.java

```diff
@@ -1489,6 +1489,65 @@ public void testDoubleAllMatch() {
       assertFalse(createStream(entrySet).mapToDouble(toDouble).allMatch(i -> Math.floor(i) == i));
    }
 
+   public void testDoubleSkip() {
+      Cache<Double, String> cache = getCache(0);
+      int range = 10;
+      // First populate the cache with a bunch of values
+      IntStream.range(0, range).mapToDouble(value -> value).forEach(i -> cache.put(i, i + "-value"));
+      CacheSet<Double> keySet = cache.keySet();
+
+      for (int i = 0; i < range; i++) {
+         List<Double> res = createStream(keySet)
+                 .sorted().skip(i).collect(Collectors.toList());
```

Comment:
Need to just add a `assertEquals(i, range - res.size());` and these should be good.